app/assets/javascripts/classes/GraphSpec.ts

```diff
@@ -18,11 +18,17 @@ class GraphSpec {
     service: string;
     metric: string;
     statistic: string;
+    points: any;
+    lines: any;
+    bars: any;
 
-    constructor(service: string, metric: string, statistic: string) {
+    constructor(service: string, metric: string, statistic: string, points: any, lines: any, bars: any) {
```

Comment:
Why are all of these "any" type?  That seems very bad in this case.  Also, there are all display parameters.  I'm not sure GraphSpec is the right place for them, but I'll think more on that.